tests/e2e/specs/upe-split/shopper/shopper-bnpls-checkout.spec.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * External dependencies
+ */
+const { shopper, merchant } = require( '@woocommerce/e2e-utils' );
+import config from 'config';
+import { uiUnblocked } from '@woocommerce/e2e-utils/build/page-utils';
+/**
+ * Internal dependencies
+ */
+import { merchantWCP, shopperWCP } from '../../../utils/flows';
+import { setupProductCheckout } from '../../../utils/payments';
+
+const bnplProviders = [
+	[ 'Affirm', 'li.payment_method_woocommerce_payments_affirm', 'button' ],
+	[
+		'Afterpay/Clearpay',
+		'li.payment_method_woocommerce_payments_afterpay_clearpay',
+		'a',
+	],
+];
+
+const UPE_METHOD_CHECKBOXES = [
+	'#inspector-checkbox-control-3', // affirm
```

Comment:
IIRC (needs testing) Jest+Puppeteer allowing using XPath selectors.
Those are nontrivial, hence here is an example of trying`"//label[contains(text(),'Affirm')]/..//input"`

Current selectors are working, but subject to fail if PM list changes.

Reply:
Yep, I explored Xpath selectors but could not make them work, probably because it was at early stage of the PR and I was dealing with too many errors 😅. I'll try again, thanks for the pointer